portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,34 +26,47 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private Map<String, String> comments = new HashMap<String, String>();
-
-    @Override
-    public void init() {
-        comments.put("Mario", "Hi!");
-        comments.put("Mario's imaginary friend", "Hi! What are you going?");
-        comments.put("Mario", "Just learning how to use JSON");
-        comments.entrySet().forEach(entry->{
-            System.out.println(entry.getKey() + ": " + entry.getValue());  
-        });
-    } 
-
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String json = convertToJson(comments);
-        System.out.println("Sending string: " + json);
-        response.setContentType("application/json;");
-        response.getWriter().println(json);
-    }
+  // IMPROVE: redundently storing display names
+  private List<List<String>> comments = new ArrayList<List<String>>();
+
+  @Override
+  public void init() {
+    comments.add(new ArrayList<String>());
+    comments.add(new ArrayList<String>());
+    comments.add(new ArrayList<String>());
+    comments.get(0).add("Mario");
+    comments.get(0).add("Hi!");
+    comments.get(1).add("Mario's imaginary friend");
+    comments.get(1).add("Hi! What are you going?");
+    comments.get(2).add("Mario");
+    comments.get(2).add("Just learning how to use JSON");
+  } 
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = convertToJson(comments);
+    System.out.println("Sending string: " + json);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
 
-    public String convertToJson(Map<String, String> comments) {
-        String json = "{ \"comments\": [";
-        for(Entry<String, String> entry : comments.entrySet()){
-            json += "{\"userName\": \"" + entry.getKey()
-                + "\", \"comment\": \"" + entry.getValue() + "\"},";
-        }
-        json = json.substring(0, json.length() - 1);
-        json += "]}";
-        return json;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    comments.add(new ArrayList<String>());
```

Comment:
I think you can reuse the getCommentRow method here too. That would make it more readable.

Reply:
Done: d22eb99